### DISCUSSION TITLE
Adding more pumps

### DISCUSSION THREAD

**Discussion by wawator** [nebhead/PiTender#12] [Q&A] [answered]
Hi!
This is some great code, I made my own code here if you are interested 😉
[WaWaToR - BarBox](https://github.com/wawator/BAR-BOX-V2) 
But yours is much better, I have a question
I have a 16 relay system and 16 pump (you can never have to many cocktails 😂)
Could you tell how I can change the code to add an other 8 relays?
Thank you and great work!

**Comment by nebhead** [comment, marked as answer]
Hi @wawator - Thanks for the praise!  Yes, you can definitely add more pumps (as many as the GPIO's will allow).  I've just pushed an update last night that will allow for this.  What you will need to do is to edit the `settings.json` file to add more pump definitions.  For example, in the below, we have added pump_09 .. pump_12, however you can easily continue to add pumps.   

```
{
  "assignments": {
    "pump_02": 27,
    "pump_03": 22,
    "pump_01": 17,
    "pump_06": 25,
    "pump_07": 2,
    "pump_04": 23,
    "pump_05": 24,
    "pump_08": 3,
    "pump_09": 18,
    "pump_10": 19,
    "pump_11": 20,
    "pump_12": 21
  },
  "inventory": {
    "pump_02": "vodka",
    "pump_03": "whiskey",
    "pump_01": "rum",
    "pump_06": "tequila",
    "pump_07": "gin",
    "pump_04": "coke",
    "pump_05": "oj",
    "pump_08": "cranberry",
    "pump_09": "kahlua",
    "pump_10": "diet",
    "pump_11": "iced_tea",
    "pump_12": "empty"
  },
  "flowrate": 40
}
```

Once you've made these changes (and reset the system), you'll be able to configure all of the pumps in the admin/settings page, as well as clean the pumps.  All available drinks will be displayed on the main menu based on what you have in the pumps.  

Happy drinking!
